packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -1928,6 +2065,54 @@ function applyOptimisticUpdates_forSettings(
   return settingsByRoomId;
 }
 
+/**
+ *
+ * Applies optimistic update to channels notification settings
+ * in a stable way. It's a deep update, and remove potential `undefined` properties
+ * from the final output object because we update with a deep partial of `ChannelsNotificationSettings`.
+ *
+ * exported for unit tests only.
+ */
+export function applyOptimisticUpdates_forChannelNotificationSettings(
+  baseChannelsNotificationSettings: BaseChannelsNotificationSettings,
+  optimisticUpdates: readonly OptimisticUpdate<BaseMetadata>[]
+): ChannelsNotificationSettings {
+  const outcomingSettings = { ...baseChannelsNotificationSettings };
+
+  for (const optimisticUpdate of optimisticUpdates) {
+    switch (optimisticUpdate.type) {
+      case "update-channels-notification-settings": {
+        const incomingSettings = optimisticUpdate.settings;
+
+        for (const channelKey of keys(incomingSettings)) {
+          const key = channelKey;
+          const channelUpdates = incomingSettings[key];
+
+          if (channelUpdates && typeof channelUpdates === "object") {
```

Comment:
Out of curiosity, what does the `typeof` check here add? If it's non-nullish, would it ever not be an object?

Reply:
You're right. Thanks for catching this. I think my brain was going to strong in type check on this one